### ISSUE TITLE
No audio in recorded file (but audio in live stream) with NDI source 

### ISSUE THREAD

**Issue by coldfire99** [DistroAV/DistroAV#621]
Description
- No audio in recorded file with NDI source

Setup

Using OBS 26.1.1. (64bit) with Windows 10
Using latest NDI plugin (OBS ndi 4.9.0)

Settings
6000 kbps
Audio 320 kbits
Videoout: 1280x720
Record Format: mkv

NDI Settings
Audio Only
Source Timing
No Hardware acceleration
no Fix for Alpha blending

Streaming and Recording at the same time:

 Result:
Stream has Audio
Recording has no audio

Expected Result
Stream has Audio
Recording has Audio

Changes1:
-Set Recording format to MP4
- Set NDI source  to Network Timing and press OK
- Set NDI source to Source Timing and Press OK

Result1:
- Stream has Audio - Recording has also Audio -> OK

Changes2
- Stop everything
- Close OBS
- Open OBS
- Start to record, Start to stream

Result2:
- Stream has Audio; recording has no Audio

Expected Result2: Stream and Recoding have Audio

Changes 3:
- Stop Stream and recording
- Set NDI source  to Network Timing and press OK
- Set NDI source to Source Timing and Press OK
- Start Stream and Recording

Result3:
- Stream and mp4 Recording have Audio - OK

Change4:
- restart OBS
- Start Stream and recording

Result4:
- Stream has Audio; recording has no Audio

Expected Result4: Stream and mp4 Recording have audio

Can you please fix this issue. I use very much this Network Source Functions with NDI and also Recording and it is really bad that it is not working!

Thanks

Chris

**Comment by Trouffman** [comment]
Hey @coldfire99 

This is a known bug : https://github.com/Palakis/obs-ndi/issues/592

Check the Issue above there is a workaround

**Comment by paulpv** [comment]
Palakis installs v4.5.1 of the 
https://github.com/Palakis/obs-ndi/blob/master/installer/installer.iss
```
#define NDIRuntimeVersion "4.5.1"
  ndiRedistURL = 'https://ndi.palakis.fr/runtime/ndi-runtime-{#NDIRuntimeVersion}-Windows.exe';
```
The latest as of this moment is 4.6.1, which he has on his share:
https://ndi.palakis.fr/runtime/

You can try installing that and report back here.

I installed the full latest NewTek NDI Tools and SDK and rebooted and my problem with this went away.

**Comment by coldfire99** [comment]
Hello, I tried installing 4.6.1 Runtime but this did not change anything. There is still no audio in the recordings.

**Comment by paulpv** [comment]
Try installing the latest NewTek NDI Tools *AND/OR* SDK.
I never actually installed Palakis' 4.6.1 redist.
I installed the latest Tools and SDK and saw that I have 4.6.1 redist installed as a result.
Maybe something in the other installs got it working.

The only other thing I changed that I think could have made any difference is I changed the output format to flv, but I had tried that earlier and it did not make any difference, so I am still thinking only the updated installs made the difference.
